src/main/java/org/apache/maven/plugins/checkstyle/exec/DefaultCheckstyleExecutor.java

```diff
@@ -316,6 +258,73 @@ public URLClassLoader run()
         return checkerListener.getResults();
     }
 
+    private void setUpCheckstyleClassloader( Checker checker,
+                                             List<String> classPathStrings,
+                                             List<String> outputDirectories )
+        throws CheckstyleExecutorException
+    {
+        final List<URL> urls = new ArrayList<>( classPathStrings.size() );
```

Comment:
personal preference; but I tend not to use final on local variables where it's not required. Up to you though.